### ISSUE TITLE
Remove non-consensus critical information from MARF

### ISSUE THREAD

**Issue by jcnelson** [blockstack/blockstack-core#1206]
In SIP 005, we have the following text describing how smart contracts are evaluated:

```
Note that _other_ kinds of smart contract data (e.g., analysis data, function bodies, etc.) must also be indexed by Stacks peers. However, this data does not need to be included in a peer's commitment, is indexed via a secondary MARF structure, and the representation of that data is therefore _not_ specified as part of the Stacks protocol.
```

I agree with this sentiment, but this is not what the code currently does -- the code commits to the hash of the JSON representation of all of this data, making it "accidentally consensus-critical".

EDIT: In particular, the state root MARF commits to the following:

* The parsed contract AST (the `ContractContext` struct)
* The analysis structure (the `ContractAnalysis` struct)

This occurs in `src/chainstate/stacks/db/transactions.rs` when processing a `TransactionSmartContract` payload.  Both structs are passed into the `ClarityBlockConnection` struct's datastore.

I don't think we need to do either of these (nor do I think we want to).  The only things the state root MARF needs to commit to are things that light clients will want proofs for.  Insofar as instantiating a smart contract, I think this means the MARF only needs to commit to data variables, data maps, tokens, NFTs, and the code body text.  What do you think @kantai?

**Comment by kantai** [comment]
I believe SIP-005 is correct as is -- the MARF should commit to the contract body, and the state of persisted variables (data-vars, maps, etc). Currently in `develop`, the MARF _also_ commits to the materialized contract environment and analysis information. This should be fixed, and I've updated this issue to reflect that.

This could be readily achieved by:

1. On contract-publish, insert a Hash(contract-body) into the MARF at a key which is roughly `(contract-publisher, contract-name)`.
2. Store the materialized contract environment and analysis in a database, keyed by `(marf-root-hash, contract-publisher, contract-name)`

Unfortunately (2) will require using the sentinel value, and then rewriting with the true root-hash once a block has finished processing. The alternative is to introduce a second MARF structure which _doesn't_ get committed to (because even though analysis and materialized contract environments aren't consensus critical, they are fork-dependent, so that state must be available in the presence of forks).

**Comment by kantai** [comment]
Closed by merging #1214
